### ISSUE TITLE
[Rule proposal] blacklist functions and/or global objects

### ISSUE THREAD

**Issue by apfelbox** [eslint/eslint#7393]
**Please describe what the rule should do:**
You can provide a list of functions that aren't allowed to be used (like `window.setInterval`, etc..).

**What category of rule is this?**

[ ] Enforces code style
[ ] Warns about a potential error
[X] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
window.setInterval(doSth, 500);

function doSth()
{
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
This is a rather generic rule and could be transformed to include prevention of access to global objects or functions.
If you will, `no-console` would be a special case of this rule.

**Comment by platinumazure** [comment]
We have a few rules (some of which are fairly new to ESLint) which might be of interest to you:

* `no-restricted-globals`: Allows you to configure globals that shouldn't be used at all
* `no-restricted-properties`: Allows you to configure object/property pairs that shouldn't be used at all

Can you take a look at those and let us know if those don't meet your needs? Thanks!

**Comment by apfelbox** [comment]
Thank you for your fast reply!

`no-restricted-properties` seems to work fine for more use case. Thank you! 🎈
